pkg/controller/endpointslicemirroring/reconciler_test.go

```diff
@@ -0,0 +1,914 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpointslicemirroring
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/component-base/metrics/testutil"
+	"k8s.io/kubernetes/pkg/controller/endpointslicemirroring/metrics"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+var defaultMaxEndpointsPerSubset = int32(1000)
```

Comment:
const

Reply:
Shouldn't we test w/ a tiny size (like 13) to get a better sense of how the slicing code is working?

Reply:
Part of the way through this PR I transitioned to an approach that would result in a 1:1 mapping between EndpointSubset and EndpointSlice (potentially 1:2 if there are dual stack IPs). That means there's not a whole lot of actual slicing logic here. With that said, it is a good idea to test with a non default value here so I've added a test covering truncation if the value is set to 2. Also updated the var to a const, thanks!